WalletWasabi/Helpers/StartupHelper.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.Win32;
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Helpers
+{
+	public static class StartupHelper
+	{
+		public static bool TryModifyStartupSetting(bool isWasabiStartsWithOS)
+		{
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				return TryModifyRegistry(isWasabiStartsWithOS);
+			}
+			else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+			{
+				// Method call here
+			}
+			else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+			{
+				// Method call here
+			}
+			return false;
+		}
+
+		private static bool TryModifyRegistry(bool isWasabiStartsWithOS)
+		{
+			try
+			{
+				// This extra check is here only to eliminate warnings.
+				if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				{
+					string keyName = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run";
+					using RegistryKey? key = Registry.CurrentUser.OpenSubKey(keyName, writable: true);
+					if (isWasabiStartsWithOS)
+					{
+						string pathToExeFile = Assembly.GetExecutingAssembly().Location;
+						pathToExeFile = pathToExeFile.Remove(pathToExeFile.Length - 4);        // This part has to change if this gets released
+						pathToExeFile += ".Fluent.Desktop.exe";
+
+						key?.SetValue("WasabiWallet", pathToExeFile);
+					}
+					else
+					{
+						key?.DeleteValue("WasabiWallet");
+					}
+
+					return true;
```

Comment:
The logic is wrong here.
Even if the `key` is null it won't change anything but will return `true` which is wrong.

There should be:
```
if (key is null)
{
   return false;
}
```
Or throw NRE, as I can see you already catch that with a custom message.